### PR TITLE
drivers: npmx: switch to ADK 1.2.0

### PR DESCRIPTION
**Pull request by jaz1-nordic**
JIRA: NRFX-3697

### CODE REVIEW

File: drivers/npmx/CMakeLists.txt
Comment (wilu-nordic) [inline drivers/npmx/CMakeLists.txt:17]: Should be moved to line `11`

**Review by wilu-nordic — APPROVED** [review]

File: drivers/npmx/Kconfig
Comment (masz-nordic) [inline drivers/npmx/Kconfig:24]:
```suggestion
		bool "nPM1300"
```

File: drivers/npmx/npmx_driver.c
Comment (masz-nordic) [inline drivers/npmx/npmx_driver.c:18]: Not sure if this will work, it could conflict with upstream binding.
If problematic, it could be renamed to `nordic_npmx_npm1300` or something similar.

**Review by magp-nordic — APPROVED** [review]

File: drivers/npmx/npmx_driver.h
Comment (masz-nordic) [reply drivers/npmx/npmx_driver.h:2]: Same in all other untouched files.

Comment (jaz1-nordic) [reply drivers/npmx/npmx_driver.h:2]: Running pre-commit gives me these results.

File: samples/vbusin/sample.yaml
Comment (masz-nordic) [inline samples/vbusin/sample.yaml:24]: @nordic-piks - is it ok?

**Review by masz-nordic — APPROVED** [review]

File: samples/vbusin/sample.yaml
Comment (nordic-piks) [reply samples/vbusin/sample.yaml:24]: Yes, however it required adaptation on the side - I will do it now.

**Review by magp-nordic — APPROVED** [review]
LGTM, I am only not sure if removing eng symbol and updating licence should be in the same commit. I think it would be better if it was separated.

**Comment by nordic-segl** [comment]
Waiting for results of integration CI
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_integration/job/main/435/

**Review by nordic-segl — APPROVED** [review]

**Pull request merged**
